tests-shared/src/main/scala/org/scalajs/dom/tests/shared/SharedTests.scala

```diff
@@ -36,4 +41,42 @@ trait SharedTests {
 
   @Test final def WindowIdbTest(): AsyncResult =
     IdbTest(window.indexedDB)
+
+  @Test
+  final def ReadableStreamTest: AsyncResult = async {
+    case class Tuna(color: String)
+
+    val expectedTunas = Seq(
+        Tuna("blue"),
+        Tuna("red")
+    )
+
+    val stream = ReadableStream[Tuna](
+        new ReadableStreamUnderlyingSource[Tuna] {
+          start = { (controller: ReadableStreamController[Tuna]) =>
+            controller.enqueue(Tuna("blue"))
+            controller.enqueue(Tuna("red"))
+            controller.close()
+          }: js.Function1[ReadableStreamController[Tuna], Unit]
```

Comment:
Scala 2.11 complains about this cast, I just could not figure out how to solve it, assistance needed 😬

Reply:
Hmm, try changing the cast to 
```suggestion
          }: js.UndefOr[js.Function1[ReadableStreamController[Tuna], js.UndefOr[Unit]]]
```

Reply:
Getting two errors with that cast:
```
\scala-js-dom\tests-shared\src\main\scala\org\scalajs\dom\tests\shared\SharedTests.scala:56:66: type mismatch;
[error]  found   : org.scalajs.dom.ReadableStreamController[Tuna] => Unit
[error]  required: scala.scalajs.js.UndefOr[scala.scalajs.js.Function1[org.scalajs.dom.ReadableStreamController[Tuna],scala.scalajs.js.UndefOr[Unit]]]
[error]     (which expands to)  scala.scalajs.js.|[scala.scalajs.js.Function1[org.scalajs.dom.ReadableStreamController[Tuna],scala.scalajs.js.|[Unit,Unit]],Unit]
[error]           start = { (controller: ReadableStreamController[Tuna]) =>
[error]                                                                  ^
```
```
\scala-js-dom\tests-shared\src\main\scala\org\scalajs\dom\tests\shared\SharedTests.scala:60:12: type mismatch;
[error]  found   : scala.scalajs.js.UndefOr[scala.scalajs.js.Function1[org.scalajs.dom.ReadableStreamController[Tuna],scala.scalajs.js.UndefOr[Unit]]]
[error]     (which expands to)  scala.scalajs.js.|[scala.scalajs.js.Function1[org.scalajs.dom.ReadableStreamController[Tuna],scala.scalajs.js.|[Unit,Unit]],Unit]
[error]  required: scala.scalajs.js.UndefOr[scala.scalajs.js.Function1[org.scalajs.dom.ReadableStreamController[Tuna],scala.scalajs.js.|[Unit,scala.scalajs.js.Promise[Unit]]]]
[error]     (which expands to)  scala.scalajs.js.|[scala.scalajs.js.Function1[org.scalajs.dom.ReadableStreamController[Tuna],scala.scalajs.js.|[Unit,scala.scalajs.js.Promise[Unit]]],Unit]
[error]           }: js.UndefOr[js.Function1[ReadableStreamController[Tuna], js.UndefOr[Unit]]]
[error]            ^
```

Reply:
Hmm, I would just keep following the hints for what is "required" 😅

Reply:
I wonder how the compiler manages to get to `scala.scalajs.js.UndefOr[scala.scalajs.js.Function1[org.scalajs.dom.ReadableStreamController[Tuna],scala.scalajs.js.UndefOr[Unit]]]`, `scala.scalajs.js.UndefOr[Unit]` is basically `Unit | Unit`? This is an odd and very specific Scala 2.11 issue.

Reply:
It is strange. I'll looking it in more detail later, if you can get everything else working until then that would be great 😁

Reply:
Try wrapping the lambda with `js.defined(...)`. That should help the type inference.

Reply:
Tried these, still got errors:
```
start = js.defined({ (controller: ReadableStreamController[Tuna]) =>
  controller.enqueue(Tuna("blue"))
  controller.enqueue(Tuna("red"))
  controller.close()
}): js.UndefOr[js.Function1[ReadableStreamController[Tuna], Unit | js.Promise[Unit]]]
```
```
start = js.defined({ (controller: ReadableStreamController[Tuna]) =>
  controller.enqueue(Tuna("blue"))
  controller.enqueue(Tuna("red"))
  controller.close()
}: js.Function1[ReadableStreamController[Tuna], Unit | js.Promise[Unit]])
```
```
start = js.defined({ (controller: ReadableStreamController[Tuna]) =>
  controller.enqueue(Tuna("blue"))
  controller.enqueue(Tuna("red"))
  controller.close()
}: js.Function1[ReadableStreamController[Tuna], Unit])
```
```
start = js.defined({ (controller: ReadableStreamController[Tuna]) =>
  controller.enqueue(Tuna("blue"))
  controller.enqueue(Tuna("red"))
  controller.close()
}): js.UndefOr[js.Function1[ReadableStreamController[Tuna], Unit]]
```

Reply:
2.12 is okay with:
```
start = js.defined({ (controller: ReadableStreamController[Tuna]) =>
  controller.enqueue(Tuna("blue"))
  controller.enqueue(Tuna("red"))
  controller.close()
}): js.UndefOr[js.Function1[ReadableStreamController[Tuna], Unit | js.Promise[Unit]]]
```
But 2.11 does not accept anything 😞